crates/stackable-operator/src/builder/pod/container.rs

```diff
@@ -21,11 +25,25 @@ pub enum Error {
         source: validation::Errors,
         container_name: String,
     },
+
+    #[snafu(display(
+        "The volumeMount is clashing with an already existing volumeMount with the same mountPath but a different content. \
+            The existing volumeMount is {existing_volume_mount:?}, the new one is {new_volume_mount:?}"
+    ))]
+    ClashingVolumeMountMountPath {
+        existing_volume_mount: Box<VolumeMount>,
+        new_volume_mount: Box<VolumeMount>,
+    },
```

Comment:
We only have access to the volume, *not* the volumeMount, so we don't know the mountPath.
We could only say 'the volume "tls" clashes, bruh', which IMHO is too less information. It's not a strong opinion but I think we would be very happy to know what that volume is (emptyDir, configMap or secret-op volume) in case we get such errors.

Reply:
Actually, looking into VolumeMount it also contains
```rust
pub sub_path: Option<String>,
# or 
pub sub_path_expr: Option<String>,
```
which I would prefer to keep (as it might be totally ok as the subpaths differ(?))

Reply:
I mean I'd say the target audience for the error message are we at Stackable (a user should never see this error, it's our fault), so I think better include to much than to less information.

Reply:
> which I would prefer to keep

Sure, then let's carry individual fields in the error message instead of the whole struct.

> I mean I'd say the target audience for the error message are we at Stackable

Yes, but I would like to prepare stackable-operator and friends to be used outside of Stackable. That's why we also need to focus on what other users might expect from these errors. And reducing the errors in complexity and focusing on the important bits of information is important in that case.

Reply:
Pushed cdf5c36, where I carry individual fields in the error message instead of the whole struct.

Reply:
Actually I needed to box the 7 Strings in 8727f9a, which is a bit ugly

Reply:
Oh, I was thinking about only using 2 or 3 strings in the message, not all of them. If we _really_ want them all, we could go back to just passing the struct.

I really doubt that we need _all_ fields in the error message. That amount of detail can be provided via tracing/logging if we want to.

So bottom-line: Either use 2-3 strings and optional logging (preferred) or the complete struct.

Reply:
Agreed.

Only because the user can't possibly fix this problem, so they should get "some" detail on what happened, and then when we receive the issue, we can turn up debug logging to see all the extra detail.

Reply:
I removed the subPathEpr in e40749a, so the enum is small enough again for clippy to stop warning. IMHO we need to keep the subPath, as we are actually using it

Reply:
Implemented in a sync meeting